app/src/main/java/com/mealsmadeeasy/ui/meal/AddToPlanActivity.kt

```diff
@@ -0,0 +1,109 @@
+package com.mealsmadeeasy.ui.meal
+
+import android.app.DatePickerDialog
+import android.os.Bundle
+import android.text.format.DateFormat
+import android.widget.TextView
+import android.widget.ImageView
+import android.widget.ImageButton
+import android.widget.Spinner
+import android.widget.Button
+import android.widget.Toast
+import com.mealsmadeeasy.MealsApplication
+import com.mealsmadeeasy.R
+import com.mealsmadeeasy.data.MealStore
+import com.mealsmadeeasy.model.MealPeriod
+import com.mealsmadeeasy.ui.BaseActivity
+import com.squareup.picasso.Picasso
+import org.joda.time.DateTime
+import java.util.Calendar
+import java.util.Date
+import javax.inject.Inject
+
+class AddToPlanActivity : BaseActivity() {
+
+    @Inject lateinit var mealStore: MealStore
+    private var cal = Calendar.getInstance()
+    var chosenTime: Long = 0
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        MealsApplication.component(this).inject(this)
+
+        setContentView(R.layout.activity_add_to_plan)
+
+        val meal = mealStore.findMealById(intent.getStringExtra("meal_id"))
+        val thumbnailView = findViewById<ImageView>(R.id.add_to_plan_thumbnail)
+        val nameView = findViewById<TextView>(R.id.add_to_plan_name)
+        val servingsView = findViewById<TextView>(R.id.add_to_plan_servings)
+        val minusButton = findViewById<ImageButton>(R.id.add_to_plan_minus)
+        val plusButton = findViewById<ImageButton>(R.id.add_to_plan_plus)
+        val dateView = findViewById<TextView>(R.id.add_to_plan_date)
+        val mealSpinner = findViewById<Spinner>(R.id.add_to_plan_spinner)
+        val addButton = findViewById<Button>(R.id.add_to_plan_button)
+
+        Picasso.with(this)
+                .load(meal.thumbnailUrl)
+                .fit()
+                .centerCrop()
+                .into(thumbnailView)
+        nameView.text = meal.name
+
+        servingsView.text = "1"
+        var numServings = 1
+        minusButton.isEnabled = false
+
+        minusButton.setOnClickListener {
+            if (numServings != 1) {
+                numServings--
+                servingsView.text = numServings.toString()
+                if (numServings == 1) {
+                    minusButton.isEnabled = false
+                }
+            }
+        }
+
+        plusButton.setOnClickListener {
+            if (numServings == 1) {
+                minusButton.isEnabled = true
+            }
+            numServings++
+            servingsView.text = numServings.toString()
+        }
+
+        setupDatePicker(dateView)
+
+        addButton.setOnClickListener {
```

Comment:
Let's also make this close the activity when you add it to the meal plan (chances are you don't want to repeat it a lot in the week). You can do this by calling `activity.finish()`.